ctru-rs/examples/title-info.rs

```diff
@@ -0,0 +1,41 @@
+use ctru::prelude::*;
+use ctru::services::am::Am;
+use ctru::services::fs::FsMediaType;
+
+fn main() {
+    ctru::use_panic_handler();
+
+    let gfx = Gfx::init().expect("Couldn't obtain GFX controller");
+    let hid = Hid::init().expect("Couldn't obtain HID controller");
+    let apt = Apt::init().expect("Couldn't obtain APT controller");
+    let am = Am::init().expect("Couldn't obtain AM controller");
+    let _console = Console::init(gfx.top_screen.borrow_mut());
+
+    let title_count = am
+        .get_title_count(FsMediaType::Sd)
+        .expect("Failed to get title count");
+    println!("This 3DS has {title_count} titles on its SD Card:");
+
+    let title_list = am
+        .get_title_list(FsMediaType::Sd)
+        .expect("Failed to get title list");
+    for id in title_list {
```

Comment:
I have a lot of downloaded titles (e-shop shutdown panic) so this loop actually covers the original message. Furthermore, it's not really useful (nor fun) for a user to get listed a bunch of hex numbers. Can we retrieve the title's names instead?

Reply:
> rust3ds is my first time doing any kind of homebrew for 3ds

Well, we always encourage using just devkitpro's toolchain first since it gives a better idea of how the system is going to actually handle the requests.

> I tried to get the actual names and couldn't figure it out on my own

Either way, let's at least not use the newline (just `print!()`with a comma after each title?)

Reply:
I'm thinking that implementing some basic scrolling would be good enough, as in just having offset variable, changing that offset with dpad/circle pad and on every change iterate through `title_list.skip(offset)`. That would also allow displaying more info of selected title id on bottom display

Reply:
I changed the interface quite a bit, what do you think now?

Reply:
> Can we retrieve the title's names instead?
In `feature/smdh` branch of my fork I implemented basic SMDH reading support and added it to `title-info` example.
I'll open PR with it after this one is merged.